### ISSUE TITLE
Add support for Cucumber Expressions

### ISSUE THREAD

**Issue by aslakhellesoy** [cucumber/cucumber-js#622]
## Summary

[Cucumber Expressions](http://docs.cucumber.io/cucumber-expressions/) is a new alternative to
Regular Expressions, which is easier to read and write, and less daunting.
## Expected Behavior

Step Defintions defined with a `String` (and not a `Regexp`) should be interpreted as a Cucumber Expression:

``` JavaScript
Given("I have {n:int} cukes in my belly", function(cukeCount) {
  // It's already converted to a number
  assert.equal(typeof cukeCount, 'number')
})
```

Likewise, if a Regular Expression is used (old school):

``` javascript
Given(/I have (\d+) cukes in my belly/, function(cukeCount) {
  // It's already converted to a number
  assert.equal(typeof cukeCount, 'number')
})
```

This is possible because the `cucumber-expression` library's `RegExp` support takes advantage of type conversions too.

Snippets for undefined steps should use Cucumber Expressions, and not Regular Expressions. We want users to start migrating, and if we can avoid it we'll not generate snippets with regular expressions. (Well, we might if enough people want it). The `cucumber-expression` library can already generate expressions from text.

Finally, the current `Transform` logic should be deprecated and the logic built into the `cucumber-expression` library should be used instead.
## Context & Motivation

Having to use Regular Expressions is one of the most common complaints against Cucumber. Other clones of Cucumber who support something similar to Cucumber Expressions seem to have happier users!

**Comment by charlierudolph** [comment]
This is a breaking change as cucumber-js currently does not do the automatic type conversions for regexps and currently supports a different version of step definitions using [strings](https://github.com/cucumber/cucumber-js/blob/8309f4487afe128b7b3a8ad9a6e6f90f5d0b06db/README.md#strings-instead-of-regular-expressions). I'll add this into my 2.0 branch

**Comment by charlierudolph** [comment]
Available in `2.0.0-rc.1`
